ansible/module_utils/hashivault.py

```diff
@@ -52,12 +54,17 @@ def hashivault_auth(client, params):
     authtype = params.get('authtype')
     username = params.get('username')
     password = params.get('password')
+    secret_id = params.get('secret_id')
+    role_id = params.get('role_id')
+
     if authtype == 'github':
         client.auth_github(token)
     elif authtype == 'userpass':
         client.auth_userpass(username, password)
     elif authtype == 'ldap':
         client.auth_ldap(username, password)
+    elif authtype == 'approle':
+        client = AppRoleClient(client,role_id,secret_id)
```

Comment:
I'm struggling to understand why this call couldn't just be something like:
```
resp = client.auth_approle(role_id,secret_id)
client.token = str(resp['auth']['client_token'])
```

Reply:
Approle tokens can be configured to be single use. Since hashivault_write does a read and write it needs to use 2 tokens. This a clean way to set the token on every call without having to modify any of the modules.

Reply:
>I'm struggling to understand what the read and write file modules do that can't be done like:

>    \- copy: 
        &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;content: "{{lookup('hashivault', 'giant', 'foo') | b64encode }}"
        &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;dest: './out.file'

I want to force base64 encoding for files so any file written or read with the module is assumed to be base64 encoded, and won't need additional validation. I also want to abstract users from worrying about encoding altogether.

Reply:
> It would be useful if the approle work was in another pr.

I can split this into separate pull requests tomorrow.

Reply:
I *think* under the covers ansible is setting up a new authorized client for every request anyway.  I think the context is thrown away for each task (I'd need to verify that).